validation_helper.go

```diff
@@ -0,0 +1,682 @@
+	}
+	return IntMaxItemsValidator{definition}, nil
+}
+
+func (i IntMaxItemsValidator) Validate(input []int) *IntMaxItemsValidationError {
+	if len(input) <= i.definition.MaxItems {
+		return nil
+	}
+	return &IntMaxItemsValidationError{
+		i.definition,
+		input,
+	}
+}
+
+type FloatMaxItemsValidator struct {
+	definition FloatMaxItemsValidatorDefinition
+}
+
+type FloatMaxItemsValidatorDefinition struct {
+	MaxItems int `json:"max_items"`
+}
+
+type FloatMaxItemsValidationError struct {
+	Definition FloatMaxItemsValidatorDefinition `json:"definition"`
+	Input      []float64                        `json:"input"`
+}
+
+func NewFloatMaxItemsValidator(definition FloatMaxItemsValidatorDefinition) (FloatMaxItemsValidator, error) {
+	if definition.MaxItems < 0 {
+		return FloatMaxItemsValidator{}, NoLengthError{"the value of maxItems should be greater than, or equal to, 0"}
+	}
+	return FloatMaxItemsValidator{definition}, nil
+}
+
+func (i FloatMaxItemsValidator) Validate(input []float64) *FloatMaxItemsValidationError {
+	if len(input) <= i.definition.MaxItems {
+		return nil
+	}
+	return &FloatMaxItemsValidationError{
+		i.definition,
+		input,
+	}
+}
+
+type StringMaxItemsValidator struct {
+	definition StringMaxItemsValidatorDefinition
+}
+
+type StringMaxItemsValidatorDefinition struct {
+	MaxItems int `json:"max_items"`
+}
+
+type StringMaxItemsValidationError struct {
+	Definition StringMaxItemsValidatorDefinition `json:"definition"`
+	Input      []string                          `json:"input"`
+}
+
+func NewStringMaxItemsValidator(definition StringMaxItemsValidatorDefinition) (StringMaxItemsValidator, error) {
+	if definition.MaxItems < 0 {
+		return StringMaxItemsValidator{}, NoLengthError{"the value of maxItems should be greater than, or equal to, 0"}
+	}
+
+	return StringMaxItemsValidator{definition}, nil
+}
+
+func (i StringMaxItemsValidator) Validate(input []string) *StringMaxItemsValidationError {
+	if len(input) <= i.definition.MaxItems {
+		return nil
+	}
+	return &StringMaxItemsValidationError{
+		i.definition,
+		input,
+	}
+}
+
+type FormatValidator struct {
+	definition FormatValidatorDefinition
+}
+
+type FormatValidatorDefinition struct {
+	Format string `json:"format"`
+}
+
+type FormatValidationError struct {
+	Definition FormatValidatorDefinition `json:"definition"`
+	Input      string                    `json:"input"`
+}
+
+func NewFormatValidator(definition FormatValidatorDefinition) (FormatValidator, error) {
+	switch definition.Format {
+	case "date-time":
+		return FormatValidator{definition}, nil
+	case "email":
+		return FormatValidator{definition}, nil
+	case "hostname":
+		return FormatValidator{definition}, nil
+	case "uri":
+		return FormatValidator{definition}, nil
+	}
+	return FormatValidator{}, InvalidFormatError{"the format is not found"}
+}
+
+func (f FormatValidator) Validate(input string) *FormatValidationError {
+	const (
```

Comment:
Initialize constances at the head of this file.
This code initialize regexp instances each time at being called `Validate` method.